consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -75,13 +75,16 @@ impl PersistentSafetyStorage {
         // storage.
         if let Err(diem_secure_storage::Error::KeyAlreadyExists(_)) = result {
             warn!("Attempted to re-initialize existing storage");
+            let waypoint_version: u64 = internal_store.get(WAYPOINT).map(|v| v.value)?;
+            counters::set_state(counters::WAYPOINT_VERSION, waypoint_version as i64);
```

Comment:
I think it's probably not necessary to update the counter in this case 😄

Reply:
Removed updating counters in this case.

(I'm glad I'm allowed to remove this since it makes the code less clean. Initially I was not sure I should add it. On one hand, it seems necessary to keep the waypoint version in counters and waypoint version of the internal store. On the other hand, it seems (based on the comments) this case should not happen/only happens in cluster test setting.)

Reply:
Yeah I see, although, this initialization should only be used for tests (e.g., cluster test) and they will only require this metric to be set if it's explicitly reading it, which it isn't 😄 If someone wants to read it, they'll explicitly have to add support in the future, but I don't think that's likely in the short term. Let's leave it as is.